### PR TITLE
[BUG] Detection Rule execution authorization model not outlined in security docs 

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Addresses #822. 

Previews below:
- [Create a detection rule](https://security-docs_889.docs-preview.app.elstc.co/guide/en/security/master/rules-ui-create.html)
- [Detections prerequisites and requirements](https://security-docs_889.docs-preview.app.elstc.co/guide/en/security/master/detections-permissions-section.html)

### CODE REVIEW

**Comment by nastasha-solomon** [comment]
@elasticmachine run elasticsearch-ci/docs rebuild

File: docs/detections/rules-ui-create.asciidoc [hunk 1]
```diff
@@ -75,7 +75,18 @@ scheduled run time.
...
 
 [role="screenshot"]
-image::images/all-rules.png[]
+image::images/all-rules.png[Shows the Rules page]
+
+[float]
+[[alerting-authorization-model]]
+=== Authorization
+
+Rules, including all background detection and the actions they generate are authorized using an {kib-ref}/api-keys.html[API key] associated with the last user to edit the rule. Upon creating or modifying a rule, an API key is generated for that user, capturing a snapshot of their privileges at that moment in time. The API key is then used to run all background tasks associated with the rule including detection checks and executing actions.
```
Comment (jmikell821) [inline docs/detections/rules-ui-create.asciidoc:84]:
```suggestion
Rules, including all background detection and the actions they generate, are authorized using an {kibana-ref}/api-keys.html[API key] associated with the last user to edit the rule. Upon creating or modifying a rule, an API key is generated for that user, capturing a snapshot of their privileges. The API key is then used to run all background tasks associated with the rule, including detection checks and executing actions.
```

File: docs/getting-started/detections-req.asciidoc [hunk 2]
```diff
@@ -103,7 +103,18 @@ Here is an example of a user who has the Detections feature enabled in all {kib}
 spaces:
 
 [role="screenshot"]
-image::images/sec-admin-user.png[]
+image::images/sec-admin-user.png[Shows user with the Detections feature enabled in all Kibana spaces]
+
+[float]
+[[alerting-auth-model]]
+=== Authorization
```
Comment (jmikell821) [inline docs/getting-started/detections-req.asciidoc:110]: The same changes above would apply here.

**Review by jmikell821 — COMMENTED** [review]
Just a few comments, then we'll have @spong take a look.

**Review by spong — APPROVED** [review]
Changes LGTM -- thanks for adding these for clarifying context @nastasha-solomon! 🙂

**Pull request merged**